test/testers.jl

```diff
@@ -42,6 +42,66 @@ primalapprox(x) = x
         end
     end
 
+    @testset "Inplace accumumulation: first on Array" begin
+        @testset "Correct definitions" begin
+            function ChainRulesCore.frule((_, ẋ), ::typeof(first), x::Array)
+                ẏ = InplaceableThunk(
+                    @thunk(first(ẋ)),
+                    ȧ -> ȧ + first(ẋ),  # This won't actually happen inplace
+                )
+                return first(x), ẏ
+            end
+            function ChainRulesCore.rrule(::typeof(first), x::Array{T}) where T
+                x_dims = size(x)
+                function first_pullback(ȳ)
+                    x̄_ret = InplaceableThunk(
+                        Thunk() do
+                            x̄ = zeros(T, x_dims)
+                            x̄[1]=ȳ
+                            x̄
+                        end,
+                        ā -> (ā[1] += ȳ; ā)
+                    )
+                    return (NO_FIELDS, x̄_ret)
+                end
+                return first(x), first_pullback
+            end
+
+            frule_test(first, (randn(4), randn(4)))
+            rrule_test(first, randn(), (randn(4), randn(4)))
+        end
+
+        @testset "Incorrect inplace definitions" begin
+            bad_first(value) = first(value)
```

Comment:
`bad_first` isn't bad, but the sensitivities we define for it are.
should i name it something else?

Reply:
just `_first`? or `my_first`?